the-collab-lab/tcl-70-smart-shopping-list
# Add styled components to ShareModal

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :art: UI Improvements       |
|   ✓   | :link: Update dependencies |

## Acceptance Criteria

- [x] The `ShareModal` component now correctly adapts to light and dark mode settings.
- [x] UI elements within the `ShareModal` maintain adequate contrast and visibility in both modes.

## Description

This PR focuses on enhancing the user interface of the `ShareModal` component by adding styled components that adapt to light and dark mode themes. Ensuring a consistent and accessible UI across different modes enhances user experience and adheres to modern UI design standards.

### Styled Components for Light and Dark Mode

The modal now features a set of styled components that react to the theme context. This ensures that the modal's visual elements are always visible and pleasant to interact with, regardless of the user's theme preference.

### Improved Theme Consistency

With the introduction of these theme-responsive styles, the application now maintains a consistent look and feel that resonates with the overall design system.

## Update

### Before
<img width="447" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/e00d9351-59ea-4a28-8738-d0d6434ce66e">

<img width="447" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/11337f7a-1a3d-4135-9f5d-90513e0e9150">

### After
<img width="525" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/4c5df18b-0826-4955-a196-b638a616e89a">
<img width="525" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/aa0ebf8e-afea-4c00-97ba-4ea4f16c7359">


## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin ag-design-share-list` and check that branch out with `git checkout ag-design-share-list`.
* Start the application with `npm start` and navigate to the modal.
* Toggle the theme between light and dark mode. Observe that the `ShareModal` components update their styles accordingly.
* Ensure that all text and interactive elements are clearly visible in both modes.
